// @sanity/desk-tool/lib/getIntentState.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getIntentState = getIntentState;
exports.setActivePanes = setActivePanes;

var _uuid = require("@sanity/uuid");

var _constants = require("./constants");

var state = {
  activePanes: []
};

function setActivePanes(panes) {
  state.activePanes = panes;
}
/**
 * @internal
 */


function getIntentState(intentName, params, currentState, payload) {
  var paneSegments = currentState && currentState.panes || [];
  var activePanes = state.activePanes || [];
  var editDocumentId = params.id || (0, _uuid.uuid)();
  var isTemplate = intentName === 'create' && params.template; // Loop through open panes and see if any of them can handle the intent

  for (var i = activePanes.length - 1; i >= 0; i--) {
    var pane = activePanes[i];

    if (pane.canHandleIntent && pane.canHandleIntent(intentName, params, {
      pane,
      index: i
    })) {
      var paneParams = isTemplate ? {
        template: params.template
      } : _constants.EMPTY_PARAMS;
      return {
        panes: paneSegments.slice(0, i).concat([[{
          id: editDocumentId,
          params: paneParams,
          payload
        }]])
      };
    }
  }

  return {
    intent: intentName,
    params,
    payload
  };
}
